cmd/clusteringress/istio/main.go

```diff
@@ -0,0 +1,143 @@
+	kubeconfig = flag.String("kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+)
+
+func main() {
+	flag.Parse()
+
+	loggingConfigMap, err := configmap.Load("/etc/config-logging")
+	if err != nil {
+		log.Fatalf("Error loading logging configuration: %v", err)
+	}
+	loggingConfig, err := logging.NewConfigFromMap(loggingConfigMap)
+	if err != nil {
+		log.Fatalf("Error parsing logging configuration: %v", err)
+	}
+	logger, atomicLevel := logging.NewLoggerFromConfig(loggingConfig, component)
+	defer logger.Sync()
+
+	// set up signals so we handle the first shutdown signal gracefully
+	stopCh := signals.SetupSignalHandler()
+
+	cfg, err := clientcmd.BuildConfigFromFlags(*masterURL, *kubeconfig)
+	if err != nil {
+		logger.Fatalf("Error building kubeconfig: %v", err)
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		logger.Fatalf("Error building kubernetes clientset: %v", err)
+	}
+
+	sharedClient, err := sharedclientset.NewForConfig(cfg)
+	if err != nil {
+		logger.Fatalf("Error building shared clientset: %v", err)
+	}
+
+	servingClient, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		logger.Fatalf("Error building serving clientset: %v", err)
+	}
+
+	configMapWatcher := configmap.NewInformedWatcher(kubeClient, system.Namespace())
+
+	opt := reconciler.Options{
+		SharedClientSet:  sharedClient,
+		KubeClientSet:    kubeClient,
+		ServingClientSet: servingClient,
+		ConfigMapWatcher: configMapWatcher,
+		Logger:           logger,
+		ResyncPeriod:     10 * time.Hour, // Based on controller-runtime default.
+		StopChannel:      stopCh,
+	}
+
+	servingInformerFactory := informers.NewSharedInformerFactory(servingClient, opt.ResyncPeriod)
+	sharedInformerFactory := sharedinformers.NewSharedInformerFactory(sharedClient, opt.ResyncPeriod)
+
+	clusterIngressInformer := servingInformerFactory.Networking().V1alpha1().ClusterIngresses()
+	virtualServiceInformer := sharedInformerFactory.Networking().V1alpha3().VirtualServices()
+	gatewayInformer := sharedInformerFactory.Networking().V1alpha3().Gateways()
+
+	// Build all of our controllers, with the clients constructed above.
+	// Add new controllers to this array.
+	controller := clusteringress.NewController(
+		opt,
+		clusterIngressInformer,
+		virtualServiceInformer,
+		gatewayInformer,
+	)
+
+	// Watch the logging config map and dynamically update logging levels.
+	opt.ConfigMapWatcher.Watch(logging.ConfigMapName(), logging.UpdateLevelFromConfigMap(logger, atomicLevel, component))
+
+	// These are non-blocking.
+	servingInformerFactory.Start(stopCh)
+	sharedInformerFactory.Start(stopCh)
+	if err := configMapWatcher.Start(stopCh); err != nil {
+		logger.Fatalf("Failed to start configuration manager: %v", err)
+	}
+
+	// Wait for the caches to be synced before starting controllers.
+	logger.Info("Waiting for informer caches to sync")
```

Comment:
We've updated this code to a more streamlined model, PTAL in `controller/main.go`.

Reply:
Upated to match main controller